File: Leetcode/Biweely Contest 50/BWC50-C.py
class Solution(object):
    def getMaximumXor(self, nums, maximumBit):
        xsum = 0
        xorsumList = []
        for x in nums:
            xsum ^= x
            xorsumList.append(xsum)
            
        bits = (1<<maximumBit)-1
        return [((~(xorsum & bits)) & bits) for xorsum in reversed(xorsumList)]

File: Leetcode/Biweely Contest 50/BWC50-C.cpp
class Solution {
public:
	vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
		int xorsum = 0;
		for (int i = 0; i < nums.size(); i++) 
			xorsum ^= nums[i];
		
		int bits = (1 << maximumBit)-1;

		vector<int> res;
		for (int i = nums.size() - 1; i >= 0; i--)
		{
			res.push_back((~(xorsum & bits)) & bits);
			xorsum ^= nums[i];
		}
        return res;
	}
};